### ISSUE TITLE
EDTA TE_purifier running for a long time

### ISSUE THREAD

**Issue by yunur13** [oushujun/EDTA#124]
Hi Shujun ,

I have run LTR, TIR , Helitron step successfully using EDTA_raw.pl script on my 5.4G genome. However, when I try to run EDTA.pl to finish the rest analysis, it has been running the following step for 48 hours
`Perform EDTA advcance filtering for raw TE candidates and generate the stage 1 library:`

I have run EDTA with the following command
`perl /EDTA/EDTA.pl --species others --step all --sensitive 0 --anno 0 --overwrite 0  --genome $genome.fasta -t 24`
The task currently running is the EDTA/util/TE_purifier.pl script, many blastn are running but nothing is written.

The following files have been generated in the $genome.fasta.mod.EDTA.combine folder:
```
$genome.fasta.mod.Helitron.raw.fa
$genome.fasta.mod.LTR.raw.fa
$genome.fasta.mod.LTR.raw.fa.cat.gz
$genome.fasta.mod.LTR.raw.fa-$genome.fasta.mod.TIR.raw.fa.fa
$genome.fasta.mod.LTR.raw.fa-$genome.fasta.mod.TIR.raw.fa.stat
$genome.fasta.mod.LTR.raw.fa.masked
$genome.fasta.mod.LTR.raw.fa.nhr
$genome.fasta.mod.LTR.raw.fa.nin
$genome.fasta.mod.LTR.raw.fa.nsq
$genome.fasta.mod.LTR.raw.fa.out
$genome.fasta.mod.LTR.raw.fa.tbl
$genome.fasta.mod.TIR.raw.fa
$genome.fasta.mod.TIR.raw.fa.nhr
$genome.fasta.mod.TIR.raw.fa.nin
$genome.fasta.mod.TIR.raw.fa.nsq
```

and `$genome.fasta.mod.LTR.raw.fa-$genome.fasta.mod.TIR.raw.fa.fa` is empty

I just wanted to check if this is normal, and if there is anything I can do to speed this up?

Many thanks

Yun

**Comment by oushujun** [comment]
Hi Yun,

You may test your EDTA on a small genome to make sure it runs smoothly. A 5.4G genome is not small and it should take a longer time.

Best,
Shujun

**Comment by yunurzzz** [comment]
Thanks for the quick reply. I have tested EDTA on Arabidopsis genome and it runs smoothly.
I will wait for its finishing.

**Comment by oushujun** [comment]
Hi Yun,

Have you finished your run?

Shujun

**Comment by yunurzzz** [comment]
Hi Shujun ,

Thanks for your reply !

The task isn't finished yet, it's still running on TE_purifier.pl, and probably stuck in blast.
Six of the same blast commands in the subprocess have been running for about two weeks, but have no output.
`blastn -db $genome.fasta.mod.LTR.raw.fa -query /dev/fd/63 -outfmt 6 -word_size 7 -evalue 1e-5 -dust no`
I've tested EDTA.pl script several times with this genome, it's always stuck here.

What should I do to make the program run smoothly?

Many thanks

Yun

**Comment by oushujun** [comment]
Hi Yun,

Sorry that your run is not complete. You may try two things: 1. update EDTA to v1.9.4. 2. Run it on an exclusive node without the intervention of other programs. `blastn` may stuck if other programs are taking over all the CPU allocations. 3. try to run EDTA from a container (ie, docker or singularity).

Best,
Shujun

**Comment by yunurzzz** [comment]
Hi Shujun,

Thank you for the tips! The task finished!

Thanks again!

Yun

**Comment by oushujun** [comment]
Hi Yun,

Can you share what did you do to finish the task eventually? People may benefit from your experience. Thanks!

Shujun

**Comment by yunurzzz** [comment]
Hi Shujun,

It is possible that there’s too much data and not enough resources. I used a node with more computing resources to complete the task.

Yun

**Comment by yuzhenpeng** [comment]
Hi,
I got the same ploblem when I run EDTA.pl to the analysis, it has been running the following step for 7 days
`Perform EDTA advcance filtering for raw TE candidates and generate the stage 1 library:`

I have run EDTA with the following command
perl /EDTA/EDTA.pl --species others --step all --sensitive 1 --anno 1 --overwrite 0 --genome $genome.fasta -t 50

**Comment by oushujun** [comment]
Hi, what version are you using? And how large is your genome?

Thanks, Shujun

**Comment by yuzhenpeng** [comment]
Hello，Shujun

> Hi, what version are you using? And how large is your genome? Thanks, Shujun


The EDTA version is v1.9.4. My mammalian genome size is about 2.5 Gb.

It seems error in this step.
`perl $EDTA_process -genome $genome -ltr $genome.EDTA.raw/$genome.LTR.raw.fa -tir $genome.EDTA.raw/$genome.TIR.raw.fa -helitron $genome.EDTA.raw/$genome.Helitron.raw.fa -repeatmasker $repeatmasker -blast $blastplus -threads $threads -protlib $protlib`

The -protlib is defined as alluniRefprexp082813 of plant genome in the database. But my genome is mammal.

**Comment by oushujun** [comment]
An easy fix will be updating to 1. 9. 9 and rerun from original folder. You
can keep EDTA_raw and delete other steps.

Shujun

**Comment by yuzhenpeng** [comment]
Hi, Shujun
perl EDTA.pl --step filter, correct?

**Comment by oushujun** [comment]
No, You don't need to specify anything. It will start from where necessary
automatically. Be sure to remove --overwrite 1 because that will delete all
existing results. You may also want --anno 1 for whole genome annotations.

Shujun

**Comment by yuzhenpeng** [comment]
I found the version of EDTA in github is 1.9.6. How to updating to 1.9.9.

Thank you

**Comment by oushujun** [comment]
The github release is 1.9.6 and the github main branch version is 1.9.9. You can simply `git clone https://github.com/oushujun/EDTA.git` to your local computer and use the existing conda env (1.9.4) to run the new version.

Shujun

**Comment by yuzhenpeng** [comment]
thanks for your kind help
